src/test/java/com/redhat/pantheon/servlet/DefaultQueryServletTest.java

```diff
@@ -0,0 +1,203 @@
+package com.redhat.pantheon.servlet;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.sling.resourcebuilder.api.ResourceBuilder;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Iterator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+public class DefaultQueryServletTest {
+
+    private final SlingContext slingContext = new SlingContext(ResourceResolverType.JCR_OAK);
+
+    @Test
+    @DisplayName("Test a simple query")
+    public void testSimpleQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
+
+        // When
+        servlet.doGet(slingContext.request(), slingContext.response());
+
+        // Then
+        JsonNode jsonNode = new ObjectMapper().readTree(slingContext.response().getOutputAsString());
+        assertEquals(10L, jsonNode.get("size").asLong());
+        assertEquals(10L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
```

Comment:
I really like this Given-When-Then pattern. It makes the tests very easy to read and understand. Thanks for setting this good example!